core/src/main/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreakerRpcClient.java

```diff
@@ -123,26 +125,27 @@ public static CircuitBreakerRpcClientBuilder builder(
     /**
      * Creates a new instance that decorates the specified {@link RpcClient}.
      */
-    CircuitBreakerRpcClient(RpcClient delegate, CircuitBreakerMapping mapping,
-                            CircuitBreakerRuleWithContent<RpcResponse> ruleWithContent) {
-        super(delegate, mapping, requireNonNull(ruleWithContent, "ruleWithContent"));
+    CircuitBreakerRpcClient(RpcClient delegate, CircuitBreakerRuleWithContent<RpcResponse> ruleWithContent,
+                            CircuitBreakerClientHandler<RpcRequest> handler) {
+        super(delegate, handler, requireNonNull(ruleWithContent, "ruleWithContent"));
     }
 
     @Override
-    protected RpcResponse doExecute(ClientRequestContext ctx, RpcRequest req, CircuitBreaker circuitBreaker)
+    protected RpcResponse doExecute(ClientRequestContext ctx, RpcRequest req,
+                                    CircuitBreakerCallback callback)
             throws Exception {
         final RpcResponse response;
         try {
             response = unwrap().execute(ctx, req);
         } catch (Throwable cause) {
-            reportSuccessOrFailure(circuitBreaker, ruleWithContent().shouldReportAsSuccess(
-                    ctx, null, cause));
+            reportSuccessOrFailure(
+                    callback, ruleWithContent().shouldReportAsSuccess(ctx, null, cause), ctx, cause);
             throw cause;
         }
 
         response.handle((unused1, cause) -> {
-            reportSuccessOrFailure(circuitBreaker,
-                                   ruleWithContent().shouldReportAsSuccess(ctx, response, cause));
+            reportSuccessOrFailure(
+                    callback, ruleWithContent().shouldReportAsSuccess(ctx, null, cause), ctx, cause);
```

Comment:
Shouldn't it be?
```suggestion
                    callback, ruleWithContent().shouldReportAsSuccess(ctx, response, cause), ctx, cause);
```

Reply:
😱   Thanks for pointing this out

Reply:
Did a full scan and made sure there aren't any more unintentional changes 😉